simpa_tests/manual_tests/generate_overview.py

```diff
@@ -0,0 +1,321 @@
+# SPDX-FileCopyrightText: 2021 Division of Intelligent Medical Systems, DKFZ
+# SPDX-FileCopyrightText: 2021 Janek Groehl
+# SPDX-License-Identifier: MIT
+
+import ast
+import glob
+import importlib
+import inspect
+import logging
+import os
+import shutil
+import sys
+import zipfile
+
+import pypandoc
+import requests
+from mdutils.mdutils import MdUtils
+
+
+class GenerateOverview():
+    """
+    Runs all scripts automatically and takes the created images and compares it with reference images.
+
+    """
+
+    def __init__(self, verbose: bool = False, save_path: str = None):
+        self.verbosity = verbose
+        self.import_path = "simpa_tests.manual_tests"
+        self.current_dir = os.path.dirname(os.path.realpath(__file__)) # directory of this script, i.e ~/simpa/simpa_tests/manual_tests
+        self.file_name = os.path.basename(__file__)
+        self.reference_figures_path = os.path.join(self.current_dir, "reference_figures/")
+        if save_path == None:
+            self.save_path = os.path.join(self.current_dir, "figures/")
+        else:
+            self.save_path = save_path
+        if not os.path.isdir(self.save_path):
+            print(f"Created {self.save_path} directory")
+            os.mkdir(self.save_path)
+        self.ref_path = os.path.join(self.current_dir, "figures_ref/")
+        self.md_name = 'manual_tests_overview'
+        self.mdFile = MdUtils(file_name=self.md_name, title='<u>Overview of Manual Test Results</u>')
+        self.set_style()
+
+        # TODO fix issue in PointSourceReconstruction.py file (make it consistent with the other manual tests)
+        self.scripts_to_neglect = ["PointSourceReconstruction.py"]
+        # For testing this file one can ignore the time consuming scripts
+        """self.scripts_to_neglect += ["DelayAndSumReconstruction.py", 
+                                   "SignedDelayMultiplyAndSumReconstruction.py", "DelayMultiplyAndSumReconstruction.py",
+                                   "TimeReversalReconstruction.py", "MinimalKWaveTest.py", "ReproduceDISMeasurements.py",
+                                   "QPAIReconstruction.py", "TestLinearUnmixingVisual.py", "SegmentationLoader.py", "SimulationWithMSOTInvision.py",
+                                   "VisualiseDevices.py", "AbsorptionAndScatteringWithinHomogenousMedium.py", "ComputeDiffuseReflectance.py", 
+                                   "CompareMCXResultsWithDiffusionTheory.py", "AbsorptionAndScatteringWithInifinitesimalSlabExperiment.py"]"""
+        
+    def download_reference_images(self):
```

Comment:
Would it make sense to check if the reference is already downloaded and can just be re-used? Together with a flag if the user wants to download the newest version

Reply:
Downloading the figures is fast. Therefore this feature is not implemented yet. Current status: every time you run the code the latest figures are downloaded. But this could be a good feature to be added

Reply:
I will write a new issue for that